packages/api/src/mappings/google/activity.ts

```diff
@@ -40,23 +58,51 @@ export const mapToActivity = (
 
         if (data.dataSourceId === sourceIdActiveMinutes) {
           const seconds = convert(sum(intValues)).from("min").to("s");
-
           activity.summary = {
             ...activity.summary,
             durations: {
               ...activity.summary?.durations,
               active_seconds: seconds,
             },
           };
+          activity.activity_logs = activity.activity_logs?.map(act => {
+            const matchingActivity = data.point.find(point =>
+              matchActivityTime(act.start_time, act.end_time, parseInt(point.startTimeNanos))
+            );
+            if (matchingActivity) {
+              act = {
+                ...act,
+                durations: {
+                  ...act.durations,
+                  active_seconds: calculateActiveSeconds(act.start_time, act.end_time),
```

Comment:
are we sure this is right? just because the activity was 360 seconds long for example, I'm not sure that it means all 360 seconds were active

have you verified that the total `active_seconds` always matches the sum of session durations?

Reply:
Yea, I think the active_seconds calculation needs to be re-written.